pageserver/src/pgdatadir_mapping.rs

```diff
@@ -388,6 +387,19 @@ impl<R: Repository> DatadirTimeline<R> {
 
         Ok(result.to_keyspace())
     }
+
+    pub fn repartition(&self, lsn: Lsn) -> Result<(KeyPartitioning, Lsn)> {
+        let mut partitioning_guard = self.partitioning.lock().unwrap();
+        if partitioning_guard.1 == Lsn(0)
+            || lsn.0 - partitioning_guard.1 .0 > self.repartition_threshold
+        {
+            let keyspace = self.collect_keyspace(lsn)?;
+            let partitioning = keyspace.partition(TARGET_FILE_SIZE_BYTES);
```

Comment:
Hmm
Now, partitioning always uses `TARGET_FILE_SIZE_BYTES` as target_size and `TEST_FILE_SIZE` doesn't work as expected.

To enable it back, we should pass `target_size` it as a function argument.

Reply:
It seems to be time to move TARGET_FILE_SIZE_BYTES to config.
I have added `compaction_target_size` parameter.
Please propose some better name if you do not like it.

Also I want to notice that there is one more pending PR with per-tenant config which once again has to be rebased afer this change (and it is reasonable to make it possible to assign tenant-spocific `compaction_target_size`)